generics/debugpool/pool.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package debugpool
+
+import (
+	"fmt"
+	"runtime/debug"
+	"sync"
+
+	"github.com/mauricelam/genny/generic"
+)
+
+// elemType is the generic type for use with the debug pool.
+type elemType generic.Type
+
+// elemTypeEqualsFn allows users to override equality checks
+// for `elemType` instances.
+type elemTypeEqualsFn func(a, b elemType) bool
+
+// elemTypePool is the backing pool wrapped by the debug pool.
+type elemTypePool interface {
+	generic.Type
```

Comment:
Hm why does this embed `interface{}`?  (generic.Type is just `interface{}` alias)

Reply:
This is one of the nice bits of the forked genny version we're on. It's a way of forcing a type to conform to an interface. This decl gets stripped away during codegen.